static/src/javascripts/projects/common/modules/navigation/newSticky.js

```diff
@@ -0,0 +1,90 @@
+define([
+    'common/utils/fastdom-promise',
+    'Promise',
+    'common/utils/$',
+    'common/utils/mediator',
+    'common/utils/detect'
+], function (
+    fastdom,
+    Promise,
+    $,
+    mediator,
+    detect) {
+
+    return function () {
+        if (detect.getBreakpoint() !== 'mobile') {
+            fastdom.read(function () {
```

Comment:
Because the window of an animation frame is very small and there are many other scripts vying for it, I'd limit `fastdom.read|write`s to the bare minimum (in this case, `$adBanner|$header.height()`) and start by creating a Promise that pipes the necessary DOM elements into it.